api/kessel/inventory/v1beta2/resource_service.proto

```diff
@@ -12,6 +12,7 @@ option java_package = "org.project_kessel.api.inventory.v1beta2";
 
 message ReportResourceRequest {
   Resource resource = 1 [  json_name = "resource" ];
+  bool read_after_write_enabled = 2;
```

Comment:
I'm wondering if we should go for a more simple name for this attribute. Something that folks could immediately recognize and understand like `ensure_consistency`, `wait_for_sync`, `synchronous`, or anything else you could come up within the same vein. I'm worried the current name may come back to bite us. IMO it's important to indicate that using this feature may introduce extra latency because we are making the async processes into sync processes before the client will receive a response.

Reply:
+1, if read-after-write is not the default, maybe calling it `async` and default to true since that would be our standard process flow and you have to explicitly turn that off for read after write?

Reply:
I liked the idea of doing `async` but we can't seem to set a default value for proto3 in the rpc itself, we could make it such that if nothing is specified its async and only when setting a value or forcing false like `  bool async = 2 [(buf.validate.field).bool.const = false];` it's synchronous but its starting to feel more complicated to reason about. The nice thing with proto3 is every field is already optional so if SPs dont set anything then we can go with a default route. I think something like `bool wait_for_sync = 2;` makes sense here. As a default it will be off if nothing is set and only when set to true does the request actually wait to sync.